pkg/storage/mongo/saver.go

```diff
@@ -25,14 +28,24 @@ func (s *ModuleStore) Save(ctx context.Context, module, version string, mod []by
 	}
 
 	zipName := s.gridFileName(module, version)
-	fs := s.s.DB(s.d).GridFS("fs")
-	f, err := fs.Create(zipName)
+	db := s.client.Database(s.db)
+	bucket, err := gridfs.NewBucket(db, options.GridFSBucket())
 	if err != nil {
 		return errors.E(op, err, errors.M(module), errors.V(version))
 	}
-	defer f.Close()
 
-	numBytesWritten, err := io.Copy(f, zip)
+	uStream, err := bucket.OpenUploadStream(zipName, options.GridFSUpload())
+	if err != nil {
+		return errors.E(op, err, errors.M(module), errors.V(version))
+	}
+	defer uStream.Close()
+
+	zipBytes, err := ioutil.ReadAll(zip)
```

Comment:
Zip can be big so we don't want to load everything into memory. Can you use `io.Copy`?